jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/UploadObjectAction.kt

```diff
@@ -47,17 +44,21 @@ class UploadObjectAction(
         val row = treeTable.selectedRow
         var nodeFile: VirtualFile? = null
         if (row > 0) {
-            val path = treeTable.tree.getPathForRow(row)
+            val path = treeTable.tree.getPathForRow(treeTable.convertRowIndexToModel(row))
             val node = (path.lastPathComponent as DefaultMutableTreeNode).userObject as S3KeyNode
             nodeFile = node.virtualFile
         }
-        val chooserDialog = fileChooserFactory.createFileChooser(descriptor, project, null)
+        val chooserDialog = FileChooserFactory.getInstance().createFileChooser(descriptor, project, null)
         val filesChosen = chooserDialog.choose(project, null)
         for (fileChosen in filesChosen) {
             ApplicationManager.getApplication().executeOnPooledThread {
                 try {
                     uploadObjectAction(client, project, fileChosen, nodeFile)
                     treeTable.refresh()
+                    if (searchTextField.text.isNotEmpty()) {
```

Comment:
Not a fan of passing in UI objects into the action. This feels like it belongs as a discreet method in the refresh PR